mkosi/__init__.py

```diff
@@ -858,22 +859,29 @@ def sign_efi_binary(context: Context, input: Path, output: Path) -> None:
         find_binary("sbsign", root=context.config.tools()) is not None
     ):
         with open(output, "wb") as f:
+            cmd: list[PathString] = [
+                "sbsign",
+                "--key", context.config.secure_boot_key,
+                "--cert", context.config.secure_boot_certificate,
+                "--output", "/dev/stdout",
+            ]
+            options: list[PathString] = [
+                "--ro-bind", context.config.secure_boot_certificate, context.config.secure_boot_certificate,
+                "--ro-bind", input, input,
+            ]
+            if context.config.secure_boot_key_source.type == KeySource.Type.engine:
+                cmd += ["--engine", context.config.secure_boot_key_source.source]
+            if context.config.secure_boot_key_source.type == KeySource.Type.file or \
+                    Path(context.config.secure_boot_key).exists():
```

Comment:
Please drop the exists() check here. If the key source is configured as `file` we should fail if the file doesn't exist instead of silently not adding specific options.

Reply:
If mypy complains just add an assert

Reply:
The key might be a file even if the source is an engine, for example when using the tpm2 engine, you just pass a key file to it, so it needs to be bind mounted, otherwise it will fail

Reply:
@bluca In that case please do `isinstance(context.config.secure_boot_key, os.PathLike)`  then instead